### PR TITLE
[C-1399] Fix chromecast

### PR DESCRIPTION
**Pull request by sliptype**
### Description

Chromecast was impressively broken, I'm not sure how it ever worked tbh

* Parts of the object being passed to `loadMedia` were undefined
  * `track.uri` - this might be because of the streaming change? Fixed this by using the same logic as in `Audio` to create the streaming endpoint
  * `track.largeArtwork` - straight up doesn't exist in the codebase, where did this go from? where did it go? something something cotton-eyed joe
  * `track.artist` also didn't exist on the track object. Fixed this by following the pattern in `Audio` and using the `getUser` selector to get the owner
* Scrubber position was being updated by both the GoogleCast client and the Audio player, resulting in flickering time values
* Time also wasn't resetting when the next queued track started playing. Fixed this by resetting the global currentTime when a new track starts in `GoogleCast.tsx`

May have a follow up pr to get rid of `global.progress` in favor of a context provider

### Dragons

Changes are relatively confined to casting

### How Has This Been Tested?

Tested casting on ios and android

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/audio/GoogleCast.tsx [hunk 1]
```diff
@@ -70,12 +104,20 @@ export const useChromecast = () => {
...
+    }
+  }, [prevTrack, track])
+
   // Load media when the cast connects
   useEffect(() => {
     if (castState === CastState.CONNECTED) {
       loadCast(track, global.progress.currentTime ?? 0)
     }
-  }, [loadCast, track, castState])
+  }, [loadCast, track, prevTrack, castState])
```
Comment (raymondjacobson) [inline packages/mobile/src/components/audio/GoogleCast.tsx:120]: does this work if you replay the same song? maybe it should be using play counter

**Review by raymondjacobson — APPROVED** [review]

File: packages/mobile/src/components/audio/GoogleCast.tsx (hunk 1 above)
Comment (sliptype) [reply packages/mobile/src/components/audio/GoogleCast.tsx:120]: Ah, no it does not. Good idea, I think counter is the move

File: packages/mobile/src/components/audio/Audio.tsx [hunk 2]
```diff
@@ -350,14 +350,23 @@ export const Audio = () => {
...
       }
-      global.progress = progress
+
+      if (!isCasting) {
+        // If we aren't casting, update the progress
+        global.progress = progress
+      } else {
+        // If we are casting, only update the seekableDuration
+        // The currentTime is set via the effect in GoogleCast.tsx
+        global.progress.seekableDuration = progress.seekableDuration
```
Comment (dylanjeffers) [inline packages/mobile/src/components/audio/Audio.tsx:360]: Do we need to also set this for when we are not casting?

File: packages/mobile/src/components/audio/GoogleCast.tsx [hunk 3]
```diff
@@ -1,53 +1,89 @@
...
   const loadCast = useCallback(
-    (track, startTime) => {
-      if (client && track) {
+    async (track, startTime) => {
+      if (client && track && owner && streamingUri) {
+        const gateways = audiusBackendInstance.getCreatorNodeIPFSGateways(
+          owner.creator_node_endpoint
+        )
+
+        const imageUrl = await audiusBackendInstance.getImageUrl(
```
Comment (dylanjeffers) [inline packages/mobile/src/components/audio/GoogleCast.tsx:62]: Can't use the useTrackArtwork hook for this?

Comment (dylanjeffers) [reply packages/mobile/src/components/audio/GoogleCast.tsx:62]: I guess cause we need it in this callback cause it's one time shot?

File: packages/mobile/src/components/audio/GoogleCast.tsx [hunk 4]
```diff
@@ -1,53 +1,89 @@
...
               type: 'musicTrack',
               images: [
                 {
-                  url: track.largeArtwork
+                  url: imageUrl
                 }
               ],
               title: track.title,
-              artist: track.artist
+              artist: owner?.name
```
Comment (dylanjeffers) [inline packages/mobile/src/components/audio/GoogleCast.tsx:79]: Owner should be there yeah?

**Review by dylanjeffers — APPROVED** [review]
Sweet, and great call on counter @raymondjacobson

File: packages/mobile/src/components/audio/Audio.tsx (hunk 2 above)
Comment (sliptype) [reply packages/mobile/src/components/audio/Audio.tsx:360]: Yes, in that case we update the entire progress object. Which includes seekableDuration

File: packages/mobile/src/components/audio/GoogleCast.tsx (hunk 3 above)
Comment (sliptype) [reply packages/mobile/src/components/audio/GoogleCast.tsx:62]: I did that at first but that hook downloads the image and then returns the url. Which makes sense when caching but we only need the url in this case

File: packages/mobile/src/components/audio/GoogleCast.tsx (hunk 4 above)
Comment (sliptype) [reply packages/mobile/src/components/audio/GoogleCast.tsx:79]: Oh yup don't need to optional chain here

**Pull request merged**